### ISSUE TITLE
OTLP | Build OTLP on centos 7 

### ISSUE THREAD

**Issue by yonatan555** [open-telemetry/opentelemetry-cpp#1749]
Hi,
I am a new user with OTLP (and new with Open-source code), so there are few things that I didn't understand clearly.
I work on centos 7 Virtual machine, 
I change ON in CMakeList.txt 
I enabled: WITH_STL & WITH_ABSEIL & WITH_OTLP & WITH_PROMETHEUS & WITH_OTLP_GRPC 
and OFF: Grpc_found, CURL_FOUND, with_exmaple, build_testing OFF
, and the rest I preserved as is.

1 - am I should ON grpc found/curl found?

I ran this command with cmake, 
cmake -DCURL_DIR=/root/vcpkg/packages/curl_x64-linux/share/curl -DCURL_LIBRARY_DIR=/root/vcpkg/packages/curl_x64-linux/lib/ -DCURL_INCLUDE_DIR=/root/vcpkg/packages/curl_x64-linux/include -DgRPC_DIR=/root/vcpkg/packages/grpc_x64-linux/include/ -Dupb_DIR=/root/vcpkg/packages/upb_x64-linux/share/upb  -Dre2_DIR=/root/vcpkg/packages/re2_x64-linux/share/re2 -Dabsl_DIR=/root/vcpkg/packages/abseil_x64-linux/share/absl -Dc-ares_DIR=/root/vcpkg/packages/c-ares_x64-linux/share/c-ares .

and got this warring and after got a build error that is associated with this warning:

CMake Warning at examples/http/CMakeLists.txt:12 (target_link_libraries):
  Target "http_client" requests linking to directory
  "/root/vcpkg/packages/curl_x64-linux/lib/".  Targets may link only to
  libraries.  CMake is dropping the item.
what I should do to fix it?

Thank you a lot.

**Comment by gabswb** [comment]
See this if you want to use vcpkg maybe : 
https://github.com/open-telemetry/opentelemetry-cpp/blob/main/INSTALL.md#using-package-managers

**Comment by yonatan555** [comment]
Hi thanks for replying,
I didn't understand what are the steps that should be done,
am I just need to install vcpkg on my VM, and run: "vcpkg install opentelemetry-cpp",
then will I get the SDK, and the whole headers/libs of OTLP?

**Comment by gabswb** [comment]
You should first learn how to use CMake, see this [tuto](https://cmake.org/cmake/help/latest/guide/tutorial/index.html) 

First you are trying to build opentelemetry-cpp with CMake while providing libraries built by vcpkg, it is not very coherent.
Then it seems that you haven't disable the build of examples :
> CMake Warning at examples/http/CMakeLists.txt:12 (target_link_libraries):

Make sure to give to CMake the option `-DWITH_EXAMPLES=OFF`

Also, you should not touch CMake variables such as `<name>_FOUND`, this variable are set by CMake.

Finally, if you want to give a path to a library, you need to include the name of the library in the path, for example : 
`patht/to/lib/libtest.a`

**Comment by ThomsonTan** [comment]
> Hi thanks for replying, I didn't understand what are the steps that should be done, am I just need to install vcpkg on my VM, and run: "vcpkg install opentelemetry-cpp", then will I get the SDK, and the whole headers/libs of OTLP?

Yes, you can this, or `vcpkg install opentelemetry-cpp[otlp]`. vcpkg is supposed to handle all the dependencies automatically.

**Comment by yonatan555** [comment]
> You should first learn how to use CMake, see this [tuto](https://cmake.org/cmake/help/latest/guide/tutorial/index.html)
> 
> First you are trying to build opentelemetry-cpp with CMake while providing libraries built by vcpkg, it is not very coherent. Then it seems that you haven't disable the build of examples :
> 
> > CMake Warning at examples/http/CMakeLists.txt:12 (target_link_libraries):
> 
> Make sure to give to CMake the option `-DWITH_EXAMPLES=OFF`
> 
> Also, you should not modify CMake variables such as `<name>_FOUND`, this variable are set by CMake.
> 
> Finally, if you want to give a path to a library, you need to include the name of the library in the path, for example : `patht/to/lib/libtest.a`

thanks worked. :)

**Comment by yonatan555** [comment]
Hi first of all thanks all,
I am wondering what will be my next steps, after I succeeded to build the project, 
I want to broadcast metrics to the Prometheus server with the gRPC protocol, what needs to be done? I feel this is a mess for me.

**Comment by owent** [comment]
I think you can only export data to prometheus server with prometheus protocol(HTTP), not gRPC.

**Comment by yonatan555** [comment]
> I think you can only export data to prometheus server with prometheus protocol(HTTP), not gRPC.

ohh right, I confused with OTLP collector.

**Comment by yonatan555** [comment]
Hi, tried to compile otlp lib with one of the examples, and succeeded, I am think something with the linker,
`/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: /tmp/ccNqBHAc.o: in function `(anonymous namespace)::initMetrics()':
main.cpp:(.text+0x3d): undefined reference to `opentelemetry::v1::exporter::otlp::OtlpGrpcMetricExporterFactory::Create(opentelemetry::v1::exporter::otlp::OtlpGrpcMetricExporterOptions const&)'
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: main.cpp:(.text+0x178): undefined reference to `opentelemetry::v1::sdk::metrics::PeriodicExportingMetricReader::PeriodicExportingMetricReader(std::unique_ptr<opentelemetry::v1::sdk::metrics::PushMetricExporter, std::default_delete<opentelemetry::v1::sdk::metrics::PushMetricExporter> >, opentelemetry::v1::sdk::metrics::PeriodicExportingMetricReaderOptions const&)'
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: main.cpp:(.text+0x220): undefined reference to `opentelemetry::v1::sdk::resource::Resource::Create(opentelemetry::v1::sdk::common::AttributeMap const&, std::string const&)'
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: main.cpp:(.text+0x242): undefined reference to `opentelemetry::v1::sdk::metrics::MeterProvider::MeterProvider(std::unique_ptr<opentelemetry::v1::sdk::metrics::ViewRegistry, std::default_delete<opentelemetry::v1::sdk::metrics::ViewRegistry> >, opentelemetry::v1::sdk::resource::Resource)'
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: main.cpp:(.text+0x2eb): undefined reference to `opentelemetry::v1::sdk::metrics::MeterProvider::AddMetricReader(std::shared_ptr<opentelemetry::v1::sdk::metrics::MetricReader>)'
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: /tmp/ccNqBHAc.o: in function `std::__shared_count<(__gnu_cxx::_Lock_policy)2>::__shared_count<opentelemetry::v1::sdk::metrics::MeterProvider*>(opentelemetry::v1::sdk::metrics::MeterProvider*)':
main.cpp:(.text._ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC2IPN13opentelemetry2v13sdk7metrics13MeterProviderEEET_[_ZNSt14__shared_countILN9__gnu_cxx12_Lock_policyE2EEC5IPN13opentelemetry2v13sdk7metrics13MeterProviderEEET_]+0x56): undefined reference to `opentelemetry::v1::sdk::metrics::MeterProvider::~MeterProvider()'
/opt/rh/devtoolset-9/root/usr/libexec/gcc/x86_64-redhat-linux/9/ld: /tmp/ccNqBHAc.o: in function `std::_Sp_counted_ptr<opentelemetry::v1::sdk::metrics::MeterProvider*, (__gnu_cxx::_Lock_policy)2>::_M_dispose()':
main.cpp:(.text._ZNSt15_Sp_counted_ptrIPN13opentelemetry2v13sdk7metrics13MeterProviderELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv[_ZNSt15_Sp_counted_ptrIPN13opentelemetry2v13sdk7metrics13MeterProviderELN9__gnu_cxx12_Lock_policyE2EE10_M_disposeEv]+0x1d): undefined reference to `opentelemetry::v1::sdk::metrics::MeterProvider::~MeterProvider()'
collect2: error: ld returned 1 exit status
make: *** [makefile:8: main] Error 1
`

I ran this command:
`g++ -o main -I./  -L./ -lopentelemetry_common -lopentelemetry_exporter_in_memory -lopentelemetry_exporter_ostream_logs -lopentelemetry_exporter_ostream_span -lopentelemetry_otlp_recordable -lopentelemetry_exporter_ostream_metrics_deprecated -lopentelemetry_http_client_curl -lopentelemetry_exporter_otlp_grpc -lopentelemetry_exporter_otlp_grpc_client -lopentelemetry_exporter_otlp_grpc_log  -lopentelemetry_exporter_otlp_http_client -lopentelemetry_exporter_otlp_http -lopentelemetry_exporter_otlp_http_log -lopentelemetry_logs -lopentelemetry_metrics_deprecated -lopentelemetry_proto                -lopentelemetry_resources -lopentelemetry_trace -lopentelemetry_version -lprometheus_exporter_deprecated    main.cpp
`
